render.go

```diff
@@ -72,24 +78,34 @@ func (r *Render) renderWindowTooSmall() {
 }
 
 func (r *Render) renderCompletion(buf *Buffer, completions *CompletionManager) {
-	max := completions.max
-	if max > r.row {
-		max = r.row
+	windowHeight := len(completions.tmp)
+	if windowHeight > int(completions.max) {
+		windowHeight = int(completions.max)
+	}
+	contentHeight := len(completions.tmp)
+
+	fractionVisible := float64(windowHeight) / float64(contentHeight)
+	fractionAbove := float64(completions.verticalScroll) / float64(contentHeight)
+
+	scrollbarHeight := int(math.Min(float64(windowHeight), math.Max(1, float64(windowHeight)*fractionVisible)))
+	scrollbarTop := int(float64(windowHeight) * fractionAbove)
+
+	isScrollThumb := func(row int) bool {
+		return scrollbarTop <= row && row <= scrollbarTop+scrollbarHeight
 	}
 
 	suggestions := completions.GetSuggestions()
 	if l := len(completions.GetSuggestions()); l == 0 {
 		return
-	} else if l > int(max) {
-		suggestions = suggestions[:max]
 	}
 
 	formatted, width := formatSuggestions(
 		suggestions,
-		int(r.col)-len(r.prefix),
+		int(r.col)-len(r.prefix)-1,
 	)
+	formatted = formatted[completions.verticalScroll : completions.verticalScroll+windowHeight]
 	l := len(formatted)
-	r.prepareArea(l)
+	r.prepareArea(windowHeight)
 
 	d := (len(r.prefix) + len(buf.Document().TextBeforeCursor())) % int(r.col)
```

Comment:
Please rewrite this line to following code.

```
// +1 means a width of scrollbar.
d := (len(r.prefix) + len(buf.Document().TextBeforeCursor()) + 1) % int(r.col)
```

<img width="627" alt="2018-02-12 18 44 12" src="https://user-images.githubusercontent.com/5564044/36090714-c2840982-1024-11e8-9fba-aef05ddf7f89.png">